vaadin-grid-pro-flow/src/main/java/com/vaadin/flow/component/gridpro/GridPro.java

```diff
@@ -410,6 +406,66 @@ public boolean getKeepEditorOpen() {
         return column;
     }
 
+    /**
+     * Event fired when the user starts to edit an existing item.
+     *
+     * @param <E> the bean type
+     */
+    @DomEvent("cell-edit-started")
+    public static class CellEditStartedEvent<E> extends ComponentEvent<GridPro<E>> {
+
+        private E item;
+        private String path;
+
+        /**
+         * Creates a new event using the given source and indicator whether the
+         * event originated from the client side or the server side.
+         *
+         * @param source     the source component
+         * @param fromClient <code>true</code> if the event originated from the client
+         * @param item       the item to be edited, provided in JSON as internally represented in Grid
+         * @param path       item subproperty that was changed
+         */
+        public CellEditStartedEvent(GridPro<E> source, boolean fromClient,
+                                        @EventData("event.detail.item") JsonObject item,
+                                        @EventData("event.detail.path") String path) {
+            super(source, fromClient);
+            this.item = source.getDataCommunicator()
+                    .getKeyMapper().get(item.getString("key"));
+            this.path = path;
+        }
+
+        /**
+         * Gets an instance of edited item.
+         *
+         * @return the instance of edited item
+         */
+        public E getItem() {
+            return item;
+        }
+
+
+        /**
+         * Gets the key of the column where item was edited.
+         *
+         * @return the key of the column
+         */
+        public String getPath() {
+            return path;
+        }
+    }
+
+    /**
+     * Registers a listener to be notified when the user starts to edit an existing item.
+     *
+     * @param listener a listener to be notified
+     * @return a handle that can be used to unregister the listener
+     */
+    public Registration addCellEditStartedListener(ComponentEventListener<CellEditStartedEvent<E>> listener) {
```

Comment:
Do we want to provide this API for users? What's the use-case? If it's needed only internally, make the classes and methods private/package-protected.

Reply:
Converted to `package-protected`.

Reply:
Why not private?

Also, the event class is still public.

Reply:
Event is available for users on the webcomponent, I guess we can make it available over here as well. Updated the docs.